.gitignore

```diff
@@ -2,9 +2,5 @@
 # will have compiled files and executables
 /target/
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
```

Comment:
I'm a bit confused about the architecture of the repository. (1) Does it implement two crates, one a library and the other a binary for generating test vectors? (2) Or does it build one crate, either the library or binary, depending on the build flag? Which approach is "better"?

I ask because the linked documentation does not say what to do for (2).

Reply:
Cargo allows for crates that have both library and binary targets. However there is a more formally supported way to do this which is a [workspace](https://doc.rust-lang.org/cargo/reference/workspaces.html). I think the idea is that we would have two crates in the workspace: one that builds the library and another that builds binaries that depend on the library crate.

I was trying to avoid doing that to keep this change simple, but I agree that `Cargo.lock` becomes ambiguous (since the recommendation is to avoid checking a library's `Cargo.lock`) and also a workspace might be more clear and better supported than the optional features I rigged up to separate binary dependencies from library dependencies. I think I'll try reorganizing this into a workspace.

Reply:
Up to you, you're much more in tune with best practice here. Just wanted to point out a potential weirdness that we may end up needing to fix one way or another.

Reply:
I'm also figuring out the best practices as I go here, so no worries. This has been a good opportunity to read up on Rust packaging. It looks like even with a workspace, you still have one big `Cargo.lock` for all the crates, but it did make dependency management cleaner, so I think the workspace is a good move.